client/core/trade.go

```diff
@@ -547,18 +548,19 @@ func (t *trackedTrade) swapMatches(matches []*matchTracker) error {
 	// Prepare the asset.Contracts.
 	swaps := new(asset.Swaps)
 	for _, match := range matches {
-		dbMatch, _, proof, _ := match.parts()
+		dbMatch, _, proof, auth := match.parts()
 		value := dbMatch.Quantity
 		if !match.trade.Sell {
 			value = calc.BaseToQuote(dbMatch.Rate, dbMatch.Quantity)
 		}
+		matchTime := encode.UnixTimeMilli(int64(auth.MatchStamp))
+		lockTime := matchTime.Add(dex.LockTimeTaker).UTC().Unix()
 		if dbMatch.Side == order.Maker {
 			proof.Secret = encode.RandomBytes(32)
 			secretHash := sha256.Sum256(proof.Secret)
 			proof.SecretHash = secretHash[:]
+			lockTime = matchTime.Add(dex.LockTimeMaker).UTC().Unix()
```

Comment:
Regarding spec clarifications, I think we should also formalize the ms -> second translation of lock time from the match time.  Seems to be a round down.